contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -479,79 +495,199 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
         reward = Math.min(CHECKPOINT_REWARD, reward);
 
         uint256 _proposerBonus = reward.mul(proposerBonus).div(100);
-        Validator storage _proposer = validators[signerToValidator[proposer]];
-        if (_proposer.contractAddress != address(0x0)) {
-            IValidatorShare(_proposer.contractAddress).addProposerBonus(_proposerBonus, _proposer.amount);
+        uint256 proposerId = signerToValidator[proposer];
+        Validator storage _proposer = validators[proposerId];
```

Comment:
we need to add proposer rewards into total rewards var.

Reply:
Where is total rewards? The one that was in the storage?

Reply:
yes